db/view/view.cc

```diff
@@ -1386,6 +1387,9 @@ future<std::optional<utils::chunked_vector<frozen_mutation_and_schema>>> view_up
     } else if (do_advance_existings) {
         co_await advance_existings();
     }
+    if (utils::get_local_injector().enter("keep_mv_read_semaphore_units") && _existing && _existing->is_clustering_row()) {
```

Comment:
Why is this injection point *here*? What does this location - part of the building of view updates from what we read - relevant to "keeping read semaphore"? Don't we get to this code **after** we already finished reading? Why is the read semaphore held while this code runs??

Reply:
I picked this location because here we finished reading the row but we're still holding the read concurrency semaphore resources - in the code before the injection we prepare the `_existing` which is a `mutation_fragment_v2` that keeps the rcs resources until it's destroyed (which happens later in `on_results()`)